google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryParameterValue.java

```diff
@@ -347,6 +356,8 @@ public static QueryParameterValue struct(Map<String, QueryParameterValue> struct
       return StandardSQLTypeName.NUMERIC;
     } else if (Date.class.isAssignableFrom(type)) {
       return StandardSQLTypeName.DATE;
```

Comment:
As you mentioned, it might be useful to also allow JSONObject in addition to a string which makes constructing json easier.